### PR TITLE
Wooden Stuffs and Armour/Tools/Weapons stats + some assets added

### PR DESCRIPTION
**Pull request by minheragon12345**

### CODE REVIEW

File: src/main/java/com/github/manasmods/tensura/handler/HurtEventHandler.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+public class HurtEventHandler {
+    @SubscribeEvent
+    public static void hurtEvent(final LivingHurtEvent e) {
+        final Entity sourceEntity = e.getSource().getDirectEntity();
+        if (sourceEntity instanceof Player player && TensuraItems.DRAGON_KNUCKLES.asItem()
+                .equals(player.getMainHandItem().getItem())) {
+            e.setAmount(e.getAmount() * 0.1F);
+            player.getMainHandItem().hurtAndBreak(1, player, Player -> {});
+        }
+    }
```
Comment (Charismara) [inline src/main/java/com/github/manasmods/tensura/handler/HurtEventHandler.java:21]: move this into the `TensuraEntityHandler` class and remove the `HurtEventHandler`

File: src/main/java/com/github/manasmods/tensura/handler/TensuraArmourHandler.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import net.minecraftforge.fml.common.Mod;
+import software.bernie.geckolib3.renderers.geo.GeoArmorRenderer;
+
+@Mod.EventBusSubscriber(modid = Tensura.MOD_ID, bus = Mod.EventBusSubscriber.Bus.MOD)
+public class TensuraArmourHandler {
+
+        @SubscribeEvent
+        public static void armourRenderer(final EntityRenderersEvent.AddLayers event) {
+            GeoArmorRenderer.registerArmorRenderer(AntiMagicMaskItem.class, new AntiMagicMaskRenderer());
+        }
```
Comment (Charismara) [inline src/main/java/com/github/manasmods/tensura/handler/TensuraArmourHandler.java:17]: do this in `TensuraClient`

File: src/main/java/com/github/manasmods/tensura/registry/ItemRegistry.java [hunk 3]
```diff
@@ -157,12 +166,35 @@ private static void registerArmour(DeferredRegister<Item> registry) {
...
+                new Item.Properties()
+                        .tab(TensuraCreativeTab.WEAPONS)
+                        .durability(1200)
+                        .fireResistant()));
         registry.register("ice_blade", () -> new IceBladeItem(TensuraToolTiers.UNIQUE,
-                9, 1));
+                9, -2.5F, new Item.Properties()
+                .tab(TensuraCreativeTab.WEAPONS)
+                .durability(1000)
+                .fireResistant()));
```
Comment (Charismara) [inline src/main/java/com/github/manasmods/tensura/registry/ItemRegistry.java:178]: don't set the ItemProperties in the registry class. Do it in the `IceBladeItem` class

File: src/main/java/com/github/manasmods/tensura/item/custom/IceBladeItem.java [hunk 4]
```diff
@@ -1,27 +1,25 @@
...
 
 import java.util.List;
 
 
-public class IceBladeItem extends SimpleSwordItem {
-    public IceBladeItem(Tier pTier, int pAttackDamageModifier, float pAttackSpeedModifier) {
-        super(pTier, pAttackDamageModifier, pAttackSpeedModifier);
+public class IceBladeItem extends SwordItem {
+    public IceBladeItem(Tier pTier, int pAttackDamageModifier, float pAttackSpeedModifier, Item.Properties pProperties) {
+        super(pTier, pAttackDamageModifier, pAttackSpeedModifier, pProperties);
```
Comment (Charismara) [inline src/main/java/com/github/manasmods/tensura/item/custom/IceBladeItem.java:21]: set the Item.Properties in the super statement and remove it from the constructor params

File: src/main/java/com/github/manasmods/tensura/registry/TensuraEntityTypes.java [hunk 5]
```diff
@@ -12,8 +13,13 @@
 
 public class TensuraEntityTypes {
 
-    public static RegistryObject<EntityType> GIANT_SALMON, GIANT_COD, ONE_EYED_OWL;
+    public static RegistryObject<EntityType> BOAT_ENTITY, GIANT_SALMON, GIANT_COD, ONE_EYED_OWL;
```
Comment (Charismara) [inline src/main/java/com/github/manasmods/tensura/registry/TensuraEntityTypes.java:16]: don't stack different types into one variable and please generify the EntityType. Example:
`public static RegistryObject<EntityType<BoatEntity>> BOAT_ENTITY;`

File: src/main/java/com/github/manasmods/tensura/registry/TensuraRegistry.java [hunk 6]
```diff
@@ -45,7 +45,7 @@ public class TensuraRegistry {
      */
     public static void register(IEventBus modEventBus) {
         BlockRegistry.register(ITEMS, BLOCKS); //Register Blocks with their BlockItems
-        ItemRegistry.register(ITEMS); //Register Items to our Registry
+        ItemRegistry.register(ITEMS); //Register Items to our Registry (wtf)
```
Comment (Charismara) [inline src/main/java/com/github/manasmods/tensura/registry/TensuraRegistry.java:48]: ??

**Review by Charismara — CHANGES_REQUESTED** [review]

File: src/main/java/com/github/manasmods/tensura/registry/TensuraRegistry.java (hunk 6 above)
Comment (minheragon12345) [reply src/main/java/com/github/manasmods/tensura/registry/TensuraRegistry.java:48]: lol sometime it just doesnt import ITEMS, dunno why, I had to keep retype the line everytime I touched ItemRegistry

File: src/main/java/com/github/manasmods/tensura/item/custom/AntiMagicMaskItem.java [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+import software.bernie.geckolib3.core.controller.AnimationController;
+import software.bernie.geckolib3.core.event.predicate.AnimationEvent;
+import software.bernie.geckolib3.core.manager.AnimationData;
+import software.bernie.geckolib3.core.manager.AnimationFactory;
+import software.bernie.geckolib3.item.GeoArmorItem;
+
+public class AntiMagicMaskItem extends GeoArmorItem implements IAnimatable {
+    private AnimationFactory factory = new AnimationFactory(this);
+    public AntiMagicMaskItem(ArmorMaterial material, EquipmentSlot slot) {
+        super(material, slot, new Properties()
```
Comment (Charismara) [inline src/main/java/com/github/manasmods/tensura/item/custom/AntiMagicMaskItem.java:25]: Please set the Material and EquipmentSlot in the super statement and remove the constructor params

File: src/main/java/com/github/manasmods/tensura/item/custom/IceBladeItem.java [hunk 8]
```diff
@@ -1,27 +1,30 @@
...
 import org.jetbrains.annotations.Nullable;
 
 import java.util.List;
 
 
-public class IceBladeItem extends SimpleSwordItem {
+public class IceBladeItem extends SwordItem {
     public IceBladeItem(Tier pTier, int pAttackDamageModifier, float pAttackSpeedModifier) {
-        super(pTier, pAttackDamageModifier, pAttackSpeedModifier);
+        super(pTier, pAttackDamageModifier, pAttackSpeedModifier,
```
Comment (Charismara) [inline src/main/java/com/github/manasmods/tensura/item/custom/IceBladeItem.java:22]: you could set the tier and int in the super statement and move the constructor params

File: src/main/java/com/github/manasmods/tensura/registry/TensuraEntityTypes.java [hunk 9]
```diff
@@ -4,16 +4,22 @@
...
 import net.minecraft.resources.ResourceLocation;
 import net.minecraft.world.entity.EntityType;
 import net.minecraft.world.entity.MobCategory;
 import net.minecraftforge.registries.DeferredRegister;
 import net.minecraftforge.registries.RegistryObject;
 
 public class TensuraEntityTypes {
-
+    public static RegistryObject<EntityType<TensuraBoatEntity>> BOAT_ENTITY;
     public static RegistryObject<EntityType> GIANT_SALMON, GIANT_COD, ONE_EYED_OWL;
```
Comment (Charismara) [inline src/main/java/com/github/manasmods/tensura/registry/TensuraEntityTypes.java:16]: Please do the same for the SALMON, COD and OWL

**Review by Charismara — CHANGES_REQUESTED** [review]

**Review by Charismara — APPROVED** [review]

**Pull request merged**